bofire/strategies/predictives/botorch.py

```diff
@@ -332,91 +360,202 @@ def _optimize_acqf_continuous(
-                num_restarts=self.num_restarts,
-                raw_samples=self.num_raw_samples,
-                equality_constraints=get_linear_constraints(
-                    domain=self.domain,
-                    constraint=LinearEqualityConstraint,
-                )
-                + interpoints,
-                inequality_constraints=get_linear_constraints(
-                    domain=self.domain,
-                    constraint=LinearInequalityConstraint,
-                ),
-                fixed_features=fixed_features,
-                nonlinear_inequality_constraints=nonlinear_constraints,  # type: ignore
-                return_best_only=True,
-                options=self._get_optimizer_options(),  # type: ignore
-                ic_generator=ic_generator,
-                **ic_gen_kwargs,
-            )
-        return candidates, acqf_vals
+        )
+        step = pd.DataFrame(sp.step(sp.start)).T
+        return pd.concat((step, self.predict(step)), axis=1)
 
-    def _ask(self, candidate_count: int) -> pd.DataFrame:  # type: ignore
-        """[summary]
+    def _ask_refill_trust_region(self, candidate_count: int) -> pd.DataFrame:
+        """If a trust region doesn't have enough experiments to reliably fit a
+        surrogate GP then we will just randomly sample from the trust region.
 
         Args:
-            candidate_count (int, optional): [description]. Defaults to 1.
+            candidate_count (int): The number of candidates to sample.
 
         Returns:
-            pd.DataFrame: [description]
+            pd.DataFrame: The candidates sampled from the trust region with
+                assigned dummy values for the predictions, standard deviations.
+        """
+        reference_experiment = self.experiments.loc[  # type: ignore
+            self.trust_region_config.X_center_idx  # type: ignore
+        ]
+        # instantiate a random strategy and ask it for the new experiments.
+        inputs = self.domain.inputs
+        outputs = self.domain.outputs
+        constraints = self.domain.constraints
+
+        # This is a bit of a hack to get the local sampler to work, ideally
+        # there would be a more elegant solution.
+        local_inputs = Inputs(
+            features=[
+                ContinuousInput(
+                    key=inp.key,
+                    bounds=(
+                        max(
+                            reference_experiment[inp.key]
+                            - self.trust_region_config.length / 2,  # type: ignore
+                            inp.lower_bound,  # type: ignore
+                        ),
+                        min(
+                            reference_experiment[inp.key]
+                            + self.trust_region_config.length / 2,  # type: ignore
+                            inp.upper_bound,  # type: ignore
+                        ),
+                    ),
+                )
+                for inp in inputs.get(ContinuousInput)
```

Comment:
Actually I think that the PR doesn't work because I've missed the logic here to actually adjust the `local_relative_bounds` of the domain based on the turbostate. 

The intent was that it would only impact the ContinuousInputs as was attempting to reuse the local search stuff. If local search is validated in that way then can also add that here

Reply:
Good point with updating the local_relative_bounds!

Reply:
And I think, you also need to something to change in line 627 where the local search is started, furtheremore also the optimize_local needs some update so that it works both for TR and LSRBO?

Reply:
I've already put a validation that you can't do both at the same time in the `BotorchStrategy` data model. In principal you might be able to remove this in the future but lets start with the minimal case

```python
    @model_validator(mode="after")
    def validate_exclusive_local_search_trust_region(self):
        # dont allow both local search and trust region to be used at the same time
        if (
            self.local_search_config is not None
            and self.trust_region_config is not None
        ):
            raise ValueError(
                "Local search and trust region optimization cannot be used at the same time."
            )
        return self
```